File: simple_1d_version/iteration1d.js
console.log("start");
let cells = [
  ["-------*-------"]["---------------"]["---------------"]["---------------"][
    "---------------"
  ]
];
let rules = {
  "***": "*",
  "**-": "*",
  "*-*": "*",
  "-**": "*",
  "---": "-",
  "-*-": "*",
  "--*": "*",
  "*--": "*"
};

let iteration1d = () => {
  let newCells = "";
  for (let i = 0; i < cells.length; i++) {
    let current = "";
    let left = "";
    let right = "";
    current = cells[i];
    i === 0 ? (left = cells[cells.length - 1]) : (left = cells[i - 1]);
    i === cells.length - 1 ? (right = cells[0]) : (right = cells[i + 1]);
    let steck = left + current + right;
    //console.log(steck);
    newCells += rules[steck];
  }
  let el = document.createElement("div");
  document.body.appendChild(el);
  el.innerHTML = newCells;
  cells = newCells;
  setTimeout(iteration1d, 400);
};
let el = document.createElement("div");
document.body.appendChild(el);
el.innerHTML = cells;
iteration1d();

console.log("end");

File: simple_1d_version/iteration1d.py
cells="----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------"
cells=cells[:len(cells)//2]*2+'*'+cells[len(cells)//2]*2

rules= {
"***": "-",
"**-": "*",
"*-*": "*",
"-**": "*",
"---": "-",
"-*-": "*",
"--*": "*",
"*--": "*"
}
newCells=""
def iteration1d(cells,newCells):
        for i in range(len(cells)):
                current=cells[i]
                left=''
                right=''
                if i==0:
                        left=cells[-1]
                        right=cells[i+1]
                elif i==len(cells)-1:
                        right=cells[0]
                        left=cells[i-1]
                else:
                        right=cells[i+1]
                        left=cells[i-1]
                steck=left+current+right
                newCells+=rules[steck]
        print(cells)
        iteration1d(newCells,'')
print(cells)
for i in range(200):
        iteration1d(cells,newCells)
